x-pack/plugins/security_solution/common/endpoint/types/file_storage.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/**
+ * The Metadata information about a file that was uploaded by Endpoint
+ * as a result of a `get-file` response action
+ */
+export interface UploadedFile {
+  file: {
+    /** The chunk size used for each chunk in this file */
+    ChunkSize?: number;
+    /**
+     * - `AWAITING_UPLOAD`: file metadata has been created. File is ready to be uploaded.
+     * - `UPLOADING`: file contents are being uploaded.
+     * - `READY`: file has been uploaded, successfully, without errors.
+     * - `UPLOAD_ERROR`: an error happened while the file was being uploaded, file contents
+     *    are most likely corrupted.
+     * - `DELETED`: file is deleted. Files can be marked as deleted before the actual deletion
+     *    of the contents and metadata happens. Deleted files should be treated as if they don’t
+     *    exist. Only files in READY state can transition into DELETED state.
+     */
+    Status: 'AWAITING_UPLOAD' | 'UPLOADING' | 'READY' | 'UPLOAD_ERROR' | 'DELETED';
+    /** File extension (if any) */
+    extension?: string;
+    hash?: {
+      md5?: string;
+      sha1?: string;
+      sha256?: string;
+      sha384?: string;
+      sha512?: string;
+      ssdeep?: string;
+      tlsh?: string;
```

Comment:
I think it's highly unlikely we will ever use these hashes. After determining the capabilities of agent/endpoint to generate hashes, we will support a specific subset, since fleet server will also need to _perform_ that same hashing, it will be a known subset we support. But is yet to be determined until they start work

md5, sha256 is probably a safe assumption. unless we need to add something simpler like crc

Reply:
will adjust once we know that.